### ISSUE TITLE
Unable to build due to spaces in file names

### ISSUE THREAD

**Issue by paroj** [paroj/gltut#65]
Originally reported by: **Anonymous**

---

I'm using Slackware Linux and GNU make to try and build the tutorials.
I can build the glsdk and the framework both in Release and Debug mode.
The tutorials fail with this kind of error:

```
#!log

==== Building Tut 08 Gimbal Lock (debug) ====
Creating obj/Debug/Tut 08 Gimbal Lock
GimbalLock.cpp
g++: error: 08: No such file or directory
g++: error: Gimbal: No such file or directory
g++: error: Lock/GimbalLock.d: No such file or directory
make[1]: *** [obj/Debug/Tut 08 Gimbal Lock/GimbalLock.o] Error 1
make: *** [Tut 08 Gimbal Lock] Error 2

```

Delving a little deeper, the problem is caused by this line (3rd) in the makefiles:

```
#!makefile

$(OBJDIR)/GimbalLock.o: GimbalLock.cpp
    @echo $(notdir $<)
    $(SILENT) $(CXX) $(CXXFLAGS) -o "$@" -MF $(@:%.o=%.d) -c "$<"
```

that produces this kind of command:

```
#!log

g++ -MMD -MP -D_CRT_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -DTIXML_USE_STL -DFREEGLUT_STATIC -DLOAD_X11 -D_LIB -DFREEGLUT_LIB_PRAGMAS=0 -DDEBUG -D_DEBUG -I../framework -I../glsdk/glload/include -I../glsdk/glimg/include -I../glsdk/glm -I../glsdk/glutil/include -I../glsdk/glmesh/include -I../glsdk/freeglut/include  -g  -o "obj/Debug/Tut 08 Gimbal Lock/GimbalLock.o" -MF obj/Debug/Tut 08 Gimbal Lock/GimbalLock.d -c "GimbalLock.cpp"

```

The problem is caused by the whitespace here: '-MF obj/Debug/Tut 08 Gimbal Lock/GimbalLock.d', adding quotes to $(@:%.o=%.d) in the Makefiles would solve the problem.

I couldn't find anyone else with this problem, but I have reproduced it in several machines.

---
- Bitbucket: https://bitbucket.org/alfonse/gltut/issue/96

**Comment by Jose123456** [comment]
The Bitbucket repo for Premake is gone. The change that fixed this problem was checked in to the Github Premake-4 repo here:

https://github.com/premake/premake-4.x/commit/a9141dee8f470f624574f4e7a0fadac20a83a0db

Unfortunately that was after the last Premake 4.4 beta was cut 5 years ago.

**Comment by Zaela24** [comment]
If anyone else runs into this issue (especially if you're using VS instead of GNU to build the project), I updated all the project files and .lua files in my fork to no longer include spaces, and added build instructions to the readme; I also added premake4 and glsdk to the main directory so that it's all in one place.

Here's the fork: https://github.com/Zaela24/gltut

**Comment by paroj** [comment]
fixed by #134
